feat(ECOM-8): implement bottom navigation bar for app sections

AhmedFayed55 opened the pull request:
feat(ECOM-8): implement bottom navigation bar for app sections
- Added CustomBottomNavigationBar with 4 items (Home, Categories, Cart, Profile)

<img width="272" height="585" alt="image" src="https://github.com/user-attachments/assets/03d875a5-816f-4f3e-b7a2-84289a597ff4" />
<img width="271" height="586" alt="image" src="https://github.com/user-attachments/assets/80af9803-04bf-4ae9-a615-11425b40ce51" />

anaa71137-beep committed:
feat(ECOM-8): implement bottom navigation bar for app sections

- Added CustomBottomNavigationBar with 4 items (Home, Categories, Cart, Profile)

john-safwat commented on lib/features/main_layout/widgets/custom_bottom_navigation_bar.dart:
this class I handling the Bottom navigation bar in bad way , just use the bottom navigation bar from material

john-safwat commented on lib/features/main_layout/main_layout.dart:
```diff
@@ -1,10 +1,33 @@
+      Center(child: Text(locale.profile)),
+    ];
+    return Scaffold(
+      bottomNavigationBar: CustomBottomNavigationBar(currentIndex: _currentIndex, onTap: (index) {
```
use the material default bottom navigation bar instead

anaa71137-beep committed:
fix(main layout):
- updated main layout to use Material BottomNavigationBar.
- applied colors and styles via Theme.

anaa71137-beep committed:
delete unused file

anaa71137-beep committed:
Delete unused file

anaa71137-beep committed:
Merge branch 'development' into feature/App-Sections-ECOM-8
